packages/@react-aria/selection/src/useSelectableItem.ts

```diff
@@ -164,6 +164,8 @@ export function useSelectableItem(options: SelectableItemOptions): SelectableIte
       }
     };
   } else if (isDisabled) {
+    // if the last focused key is the current item, keep the element focusable
+    itemProps.tabIndex = key === manager.focusedKey ? 0 : undefined;
```

Comment:
Perhaps this should also check if focus is still within the collection via `manager.isFocused` as well. This would keep the column headers non-tabbable if focus is coming from outside the table and there still isn't any content in the table, making it so a single tab keypress would still go to a link inside the illustrated message.